### PR TITLE
Update CLI tests

### PR DESCRIPTION
**Pull request by Ostrokrzew**

### CODE REVIEW

File: test/functional/tests/cli/test_cli_start_stop.py [hunk 1]
```diff
@@ -16,75 +16,192 @@
...
+                         f"No cache should be present after stopping the cache.")
         output = casadm.list_caches(shortcut=shortcut)
+        cli_messages.check_stdout_msg(output, cli_messages.no_caches_running)
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.nand, DiskType.optane]))
+@pytest.mark.parametrize("shortcut", [True, False])
+def test_cli_start_stop_custom_value(shortcut):
+    """
+        title: Test for starting cache - short and long command version
```
Comment (katlapinka) [inline test/functional/tests/cli/test_cli_start_stop.py:60]: This and previous test titles are the same. Please fix it.

File: test/functional/tests/cli/test_cli_start_stop.py [hunk 2]
```diff
@@ -16,75 +16,192 @@
...
-        TestRun.fail("Core path should equal to path of core added")
+    with TestRun.step("Remove the core from the cache."):
+        casadm.remove_core(cache.cache_id, core.core_id, shortcut=shortcut)
 
+    with TestRun.step("Check if the core is successfully removed from still running cache."):
+        caches = casadm_parser.get_caches()
+        if len(caches) != 1:
+            TestRun.fail("One cache should be present still after removing the core.")
+        if len(caches[0].get_core_devices()) != 0:
+            TestRun.fail("No core devices should be present after removing the core.")
```
Comment (katlapinka) [inline test/functional/tests/cli/test_cli_start_stop.py:136]: core device

File: test/functional/tests/cli/test_cli_start_stop.py [hunk 3]
```diff
@@ -16,75 +16,192 @@
...
+
+    with TestRun.step("Remove the core from the cache."):
+        casadm.remove_core(cache.cache_id, core.core_id, shortcut=shortcut)
+
+    with TestRun.step("Check if the core is successfully removed from still running cache."):
+        caches = casadm_parser.get_caches()
+        if len(caches) != 1:
+            TestRun.fail("One cache should be present still after removing the core.")
+        if len(caches[0].get_core_devices()) != 0:
+            TestRun.fail("No core devices should be present after removing the core.")
```
Comment (katlapinka) [inline test/functional/tests/cli/test_cli_start_stop.py:189]: core device

**Review by katlapinka — CHANGES_REQUESTED** [review]

File: test/functional/tests/cli/test_cli_start_stop.py [hunk 4]
```diff
@@ -15,76 +15,193 @@
...
-    with TestRun.LOGGER.step("Check if cache started successfully"):
+    with TestRun.step("Check if the cache has started successfully."):
         caches = casadm_parser.get_caches()
         if len(caches) != 1:
-            TestRun.LOGGER.error(f"There is wrong caches count found in OS: {len(caches)}")
-        if caches[0].cache_device.system_path != cache_device.system_path:
-            TestRun.LOGGER.error(f"Cache started using wrong device: "
-                                 f"{caches[0].cache_device.system_path}. "
-                                 f"Should be {cache_device.system_path}")
+            TestRun.fail(f"There is wrong number of caches found in the OS: {len(caches)}. "
```
Comment (Deixx) [inline test/functional/tests/cli/test_cli_start_stop.py:38]: a wrong number

File: test/functional/tests/cli/test_cli_start_stop.py [hunk 5]
```diff
@@ -15,76 +15,193 @@
...
-    with TestRun.LOGGER.step("Stop cache"):
-        casadm.stop_cache(cache_id=caches[0].cache_id, shortcut=shortcut)
+    with TestRun.step("Stop the cache."):
+        casadm.stop_cache(cache.cache_id, shortcut=shortcut)
 
-    with TestRun.LOGGER.step("Check if cache stopped properly"):
+    with TestRun.step("Check if the cache has stopped properly."):
+        caches = casadm_parser.get_caches()
+        if len(caches) != 0:
+            TestRun.fail(f"There is wrong number of caches found in the OS: {len(caches)}."
```
Comment (Deixx) [inline test/functional/tests/cli/test_cli_start_stop.py:50]: a wrong number

File: test/functional/tests/cli/test_cli_start_stop.py [hunk 6]
```diff
@@ -15,76 +15,193 @@
...
+
+    with TestRun.step("Start the cache with random ID."):
+        cache_id = randint(1, 16384)
+        cache = casadm.start_cache(cache_device, cache_id=cache_id, shortcut=shortcut, force=True)
+        TestRun.LOGGER.info(f"Cache ID: {cache_id}")
+
+    with TestRun.step("Check if the cache has started successfully."):
+        caches = casadm_parser.get_caches()
+        if len(caches) != 1:
+            TestRun.fail(f"There is wrong number of caches found in the OS: {len(caches)}. "
```
Comment (Deixx) [inline test/functional/tests/cli/test_cli_start_stop.py:80]: a wrong number

File: test/functional/tests/cli/test_cli_start_stop.py [hunk 7]
```diff
@@ -15,76 +15,193 @@
...
         if len(caches) != 1:
-            TestRun.LOGGER.error(f"There is wrong caches count found in OS: {len(caches)}")
-        if caches[0].cache_device.system_path != cache_device.system_path:
-            TestRun.LOGGER.error(f"Cache started using wrong device: "
-                                 f"{caches[0].cache_device.system_path}. "
-                                 f"Should be {cache_device.system_path}")
+            TestRun.fail(f"There is wrong number of caches found in the OS: {len(caches)}. "
+                         f"Should be only 1.")
+        if cache.cache_device.system_path != cache_device.system_path:
+            TestRun.fail(f"The cache started using wrong device: {cache.cache_device.system_path}."
```
Comment (Deixx) [inline test/functional/tests/cli/test_cli_start_stop.py:41]: has started using a wrong device

File: test/functional/tests/cli/test_cli_start_stop.py [hunk 8]
```diff
@@ -15,76 +15,193 @@
...
+        cache = casadm.start_cache(cache_device, cache_id=cache_id, shortcut=shortcut, force=True)
+        TestRun.LOGGER.info(f"Cache ID: {cache_id}")
+
+    with TestRun.step("Check if the cache has started successfully."):
+        caches = casadm_parser.get_caches()
+        if len(caches) != 1:
+            TestRun.fail(f"There is wrong number of caches found in the OS: {len(caches)}. "
+                         f"Should be only 1.")
+        if cache.cache_device.system_path != cache_device.system_path:
+            TestRun.fail(f"The cache started using wrong device: {cache.cache_device.system_path}."
```
Comment (Deixx) [inline test/functional/tests/cli/test_cli_start_stop.py:83]: has started using a wrong device

File: test/functional/tests/cli/test_cli_start_stop.py [hunk 9]
```diff
@@ -15,76 +15,193 @@
...
+          - The cache has successfully started with custom ID
+          - The cache has successfully stopped
+    """
+    with TestRun.step("Prepare the device for the cache."):
+        cache_device = TestRun.disks['cache']
+        cache_device.create_partitions([Size(500, Unit.MebiByte)])
+        cache_device = cache_device.partitions[0]
+
+    with TestRun.step("Start the cache with random ID."):
+        cache_id = randint(1, 16384)
```
Comment (Deixx) [inline test/functional/tests/cli/test_cli_start_stop.py:73]: I'd either assign those numbers to some MIN/MAX_ALLOWED_ID variables or I'd add a comment, to clarify those numbers are not picked at random, but they are significant

File: test/functional/tests/cli/test_cli_start_stop.py [hunk 10]
```diff
@@ -15,76 +15,193 @@
...
+        casadm.stop_cache(cache.cache_id, shortcut=shortcut)
+
+    with TestRun.step("Check if the cache has stopped properly."):
         caches = casadm_parser.get_caches()
         if len(caches) != 0:
-            TestRun.LOGGER.error(f"There is wrong caches count found in OS: {len(caches)}. "
-                                 f"Should be 0.")
-        if output.stdout != "No caches running":
-            TestRun.LOGGER.error("There is no 'No caches running' info in casadm -L output")
+            TestRun.fail(f"There is wrong number of caches found in the OS: {len(caches)}."
```
Comment (Deixx) [inline test/functional/tests/cli/test_cli_start_stop.py:92]: a wrong number

File: test/functional/tests/cli/test_cli_start_stop.py [hunk 11]
```diff
@@ -15,76 +15,193 @@
...
 @pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
 @pytest.mark.parametrize("shortcut", [True, False])
-def test_cli_add_remove_default_value(shortcut):
-    cache_device = TestRun.disks['cache']
-    cache_device.create_partitions([Size(50, Unit.MebiByte)])
-    cache_device = cache_device.partitions[0]
-    cache = casadm.start_cache(cache_device, shortcut=shortcut, force=True)
+def test_cli_add_remove_default_id(shortcut):
+    """
+        title: Test for adding and removing core with default ID - short and long command version
```
Comment (Deixx) [inline test/functional/tests/cli/test_cli_start_stop.py:103]: a core with a default ID

File: test/functional/tests/cli/test_cli_start_stop.py [hunk 12]
```diff
@@ -15,76 +15,193 @@
...
+        description: |
+          Start the new cache and add the core to it without passing core ID as an argument
```
Comment (Deixx) [inline test/functional/tests/cli/test_cli_start_stop.py:105]: a new cache and add a core ... a core ID

File: test/functional/tests/cli/test_cli_start_stop.py [hunk 13]
```diff
@@ -15,76 +15,193 @@
...
+          and then remove this core from the cache.
+        pass_criteria:
+          - The core is added to the cache with default ID
```
Comment (Deixx) [inline test/functional/tests/cli/test_cli_start_stop.py:108]: a default ID

File: test/functional/tests/cli/test_cli_start_stop.py [hunk 14]
```diff
@@ -15,76 +15,193 @@
...
-        TestRun.fail("One core should be present in cache")
-    if caches[0].get_core_devices()[0].core_device.system_path != core_device.system_path:
-        TestRun.fail("Core path should equal to path of core added")
+    with TestRun.step("Remove the core from the cache."):
+        casadm.remove_core(cache.cache_id, core.core_id, shortcut=shortcut)
 
+    with TestRun.step("Check if the core is successfully removed from still running cache."):
+        caches = casadm_parser.get_caches()
+        if len(caches) != 1:
+            TestRun.fail("One cache should be present still after removing the core.")
```
Comment (Deixx) [inline test/functional/tests/cli/test_cli_start_stop.py:134]: still present

File: test/functional/tests/cli/test_cli_start_stop.py [hunk 15]
```diff
@@ -15,76 +15,193 @@
...
-        TestRun.fail("No cache should be present after stopping the cache")
-    if output.stdout != "No caches running":
-        TestRun.fail(f"Invalid message, expected 'No caches running', got {output.stdout}")
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.nand, DiskType.optane]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+@pytest.mark.parametrize("shortcut", [True, False])
+def test_cli_add_remove_custom_id(shortcut):
+    """
+        title: Test for adding and removing core with custom ID - short and long command version
```
Comment (Deixx) [inline test/functional/tests/cli/test_cli_start_stop.py:154]: a core with a custom ID

File: test/functional/tests/cli/test_cli_start_stop.py [hunk 16]
```diff
@@ -15,76 +15,193 @@
...
+        description: |
+          Start the new cache and add the core to it with passing random core ID
```
Comment (Deixx) [inline test/functional/tests/cli/test_cli_start_stop.py:156]: a new cache ... a core ... a random core ID

File: test/functional/tests/cli/test_cli_start_stop.py [hunk 17]
```diff
@@ -15,76 +15,193 @@
...
+          - The core is added to the cache with default ID
+          - The core is successfully removed from the cache
+    """
+    with TestRun.step("Prepare the devices."):
+        cache_disk = TestRun.disks['cache']
+        cache_disk.create_partitions([Size(50, Unit.MebiByte)])
+        cache_device = cache_disk.partitions[0]
+        core_device = TestRun.disks['core']
+
+    with TestRun.step("Start the cache and add the core with random ID."):
```
Comment (Deixx) [inline test/functional/tests/cli/test_cli_start_stop.py:168]: a random ID

File: test/functional/tests/cli/test_cli_start_stop.py [hunk 18]
```diff
@@ -15,76 +15,193 @@
...
+        cache = casadm.start_cache(cache_device, shortcut=shortcut, force=True)
+        core = casadm.add_core(cache, core_device, core_id=core_id, shortcut=shortcut)
+        TestRun.LOGGER.info(f"Core ID: {core_id}")
+
+    with TestRun.step("Check if the core is added to the cache."):
+        caches = casadm_parser.get_caches()
+        if len(caches[0].get_core_devices()) != 1:
+            TestRun.fail("One core should be present in the cache.")
+        if caches[0].get_core_devices()[0].system_path != core.system_path:
+            TestRun.fail("The core path should equal to the path of the core added.")
```
Comment (Deixx) [inline test/functional/tests/cli/test_cli_start_stop.py:179]: be equal

File: test/functional/tests/cli/test_cli_start_stop.py [hunk 19]
```diff
@@ -15,76 +15,193 @@
...
+
+    with TestRun.step("Remove the core from the cache."):
+        casadm.remove_core(cache.cache_id, core.core_id, shortcut=shortcut)
+
+    with TestRun.step("Check if the core is successfully removed from still running cache."):
+        caches = casadm_parser.get_caches()
+        if len(caches) != 1:
+            TestRun.fail("One cache should be present still after removing the core.")
```
Comment (Deixx) [inline test/functional/tests/cli/test_cli_start_stop.py:187]: still present

File: test/functional/tests/cli/test_cli_start_stop.py [hunk 20]
```diff
@@ -15,76 +15,193 @@
...
+          - The core is successfully removed from the cache
+    """
+    with TestRun.step("Prepare the devices."):
+        cache_disk = TestRun.disks['cache']
+        cache_disk.create_partitions([Size(50, Unit.MebiByte)])
+        cache_device = cache_disk.partitions[0]
+        core_device = TestRun.disks['core']
+
+    with TestRun.step("Start the cache and add the core with random ID."):
+        core_id = randint(0, 4095)
```
Comment (Deixx) [inline test/functional/tests/cli/test_cli_start_stop.py:169]: Same as with cache id ranges

**Review by Deixx — CHANGES_REQUESTED** [review]
Mostly grammar (I saw your grammar corrections and added mine), but not only ;)

**Review by Deixx — APPROVED** [review]

**Review by karolinavelkaja — APPROVED** [review]

**Review by katlapinka — APPROVED** [review]

**Pull request merged**